src/main/java/org/mule/extensions/jms/internal/source/JmsListener.java

```diff
@@ -280,7 +284,7 @@ private void doReply(JmsResponseMessageBuilder messageBuilder,
         LOGGER.debug(format("Begin reply to destination [%s] of type [%s]", destinationName, replyToTopic ? TOPIC : QUEUE));
       }
 
-      Message message = messageBuilder.build(connection.getJmsSupport(), session.get(), config);
+      Message message = messageBuilder.build(connection.getJmsSupport(), ALWAYS, correlationInfo, session.get(), config);
```

Comment:
shouldn't this be AUTO so it checks for isOutboundCorrelationEnabled?

Reply:
yes, good catch